File: claude/nivel avanzado/python/cadena.py
def es_palindromo(texto):
    """
    Función que determina si una cadena de texto es un palíndromo,
    ignorando espacios, puntuación y diferencias entre mayúsculas y minúsculas.
    
    Args:
        texto (str): Cadena de texto a evaluar
        
    Returns:
        bool: True si es palíndromo, False en caso contrario
    """
    import re
    
    # Eliminar espacios, puntuación y convertir a minúsculas
    texto_procesado = re.sub(r'[^\w]', '', texto.lower())
    
    # Verificar si es palíndromo
    return texto_procesado == texto_procesado[::-1]

# Solicitar valor al usuario
entrada = input("Ingrese una cadena de texto para verificar si es palíndromo: ")

# Verificar y mostrar el resultado
if es_palindromo(entrada):
    print(f'"{entrada}" es un palíndromo.')
else:
    print(f'"{entrada}" no es un palíndromo.')

File: claude/nivel avanzado/javascript/cadena.js
/**
 * Función que determina si una cadena de texto es un palíndromo,
 * ignorando espacios, puntuación y diferencias entre mayúsculas y minúsculas.
 * 
 * @param {string} texto - Cadena de texto a evaluar
 * @returns {boolean} true si es palíndromo, false en caso contrario
 */
function esPalindromo(texto) {
    // Eliminar espacios, puntuación y convertir a minúsculas
    const textoProcesado = texto.toLowerCase().replace(/[^\w]/g, '').replace(/_/g, '');
    
    // Verificar si es palíndromo
    const textReversed = textoProcesado.split('').reverse().join('');
    return textoProcesado === textReversed;
}

// Solicitar valor al usuario
const entrada = prompt("Ingrese una cadena de texto para verificar si es palíndromo:");

// Verificar y mostrar el resultado
if (esPalindromo(entrada)) {
    console.log(`"${entrada}" es un palíndromo.`);
} else {
    console.log(`"${entrada}" no es un palíndromo.`);
}